griptape/engines/rag/modules/response/prompt_response_rag_module.py

```diff
@@ -11,50 +12,52 @@
 
-        if query:
-            tokenizer = self.prompt_driver.tokenizer
-            text_chunks = []
-            system_prompt = self.generate_system_template(text_chunks, before_query, after_query)
+        for artifact in context.text_chunks:
+            included_chunks.append(artifact)
 
-            for artifact in text_artifact_chunks:
-                text_chunks.append(artifact.value)
+            system_prompt = self.generate_system_template(context, included_chunks)
+            message_token_count = self.prompt_driver.tokenizer.count_tokens(
+                self.prompt_driver.prompt_stack_to_string(self.generate_query_prompt_stack(system_prompt, query))
+            )
 
-                system_prompt = self.generate_system_template(text_chunks, before_query, after_query)
-                message_token_count = self.prompt_driver.tokenizer.count_tokens(
-                    self.prompt_driver.prompt_stack_to_string(self.generate_query_prompt_stack(system_prompt, query))
-                )
+            if message_token_count + self.answer_token_offset >= tokenizer.max_input_tokens:
+                included_chunks.pop()
 
-                if message_token_count + self.answer_token_offset >= tokenizer.max_input_tokens:
-                    text_chunks.pop()
+                system_prompt = self.generate_system_template(context, included_chunks)
 
-                    system_prompt = self.generate_system_template(text_chunks, before_query, after_query)
+                break
 
-                    break
+        output = self.prompt_driver.run(self.generate_query_prompt_stack(system_prompt, query)).to_artifact()
 
-            output = self.prompt_driver.run(self.generate_query_prompt_stack(system_prompt, query)).to_artifact()
-
-            if isinstance(output, TextArtifact):
-                context.output = output
-            else:
-                raise ValueError("Prompt driver did not return a TextArtifact")
+        if isinstance(output, TextArtifact):
+            context.output = output
+        else:
+            raise ValueError("Prompt driver did not return a TextArtifact")
 
         return context
 
-    def default_system_template_generator(
-        self, text_chunks: list[str], before_system_prompt: list, after_system_prompt: list
-    ) -> str:
+    def default_system_template_generator(self, context: RagContext, artifacts: list[TextArtifact]) -> str:
         return J2("engines/rag/modules/response/prompt/system.j2").render(
-            text_chunks=text_chunks,
-            before_system_prompt="\n\n".join(before_system_prompt),
-            after_system_prompt="\n\n".join(after_system_prompt),
+            text_chunks=[c.to_text() for c in artifacts],
+            before_system_prompt="\n\n".join(context.before_query),
+            after_system_prompt="\n\n".join(context.after_query),
         )
+
+    def references_from_artifacts(self, artifacts: list[TextArtifact]) -> list[Reference]:
```

Comment:
Does this method belong in this class? It isn't used by the class and doesn't use the class' state. Seems like it belongs in utils

Reply:
Moving it to utils.

Reply:
Thanks for moving the implementation to utils, but how come you kept the method?

Reply:
Oops, I meant to call it from `RagContext` not `PromptResponseRagModule`. Fixed.